src/app/_components/shared/organisms/ProductList/ProductList.tsx

```diff
@@ -0,0 +1,32 @@
+'use client'
+import React, { useEffect, useState } from 'react'
+import { ProductListItem } from '../ProductListItem/ProductListItem'
+import styles from './ProductList.module.scss'
+import { InMemoryProductsRepository } from '@infra/products/InMemoryProductsRepository'
+import { Product } from '@core/products/entities/Product'
+
+
+export const ProductList = () => {
+  const [products, setProducts] = useState<Product[]>([])
+
+  useEffect(()=> {
+    const repository = new InMemoryProductsRepository()
+
+    repository.findAll().then((products) => {
+      setProducts(products)
+    })
+  }, [])
```

Comment:
The mechanism for fetching the stuff is okay with the useEffect and useState. But you shouldn't access to the repository directly. We need a controller layer here.